lti_consumer/plugin/views.py

```diff
@@ -379,21 +416,50 @@ def access_token_endpoint(request, lti_config_id=None, usage_id=None):
         Sucess: https://tools.ietf.org/html/rfc6749#section-4.4.3
         Failure: https://tools.ietf.org/html/rfc6749#section-5.2
     """
+    external_id = f"{external_app}:{external_slug}"
 
     try:
-        if lti_config_id:
+        if usage_id:
+            lti_config = LtiConfiguration.objects.get(location=UsageKey.from_string(usage_id))
+            version = lti_config.version
+            lti_consumer = lti_config.get_lti_consumer()
+        elif lti_config_id:
             lti_config = LtiConfiguration.objects.get(config_id=lti_config_id)
-        else:
-            usage_key = UsageKey.from_string(usage_id)
-            lti_config = LtiConfiguration.objects.get(location=usage_key)
-    except LtiConfiguration.DoesNotExist as exc:
-        log.warning("Error getting the LTI configuration with id %r: %s", lti_config_id, exc, exc_info=True)
+            version = lti_config.version
+            lti_consumer = lti_config.get_lti_consumer()
+        elif external_app and external_slug:
+            lti_config = get_external_config_from_filter({}, external_id)
+
+            if not lti_config:
+                raise ValueError("External LTI configuration not found")
+
+            version = lti_config.get("version")
+            # External LTI configurations don't have a get_lti_consumer method
+            # so we initialize the LtiConsumer1p3 class using the external config data.
+            lti_consumer = LtiConsumer1p3(
+                iss=get_lti_api_base(),
+                lti_oidc_url=None,
+                lti_launch_url=None,
+                client_id=lti_config.get("lti_1p3_client_id"),
+                deployment_id=None,
+                rsa_key=lti_config.get("lti_1p3_private_key"),
+                rsa_key_id=lti_config.get("lti_1p3_private_key_id"),
+                redirect_uris=None,
+                tool_key=lti_config.get("lti_1p3_tool_public_key"),
+                tool_keyset_url=lti_config.get("lti_1p3_tool_keyset_url"),
+            )
```

Comment:
I'm thinking about this again and wonder if we actually need to pass the external ID to these views. Wouldn't using the `lti_config_id` (UUID) instead be better? Whenever we use the external configuration, at least one `LtiConfiguration` entry always points there.

In other words:
1. We change `lti_consumer/api.py` to provide the UUID instead of the `namespace/config_id` in the URLs.
2. In this view, we use `lti_config = LtiConfiguration.objects.get(config_id=lti_config_id)` to instantiate the config from the DB.
3. We run `lti_config.get_lti_consumer()` and still get the correct `LtiConsumer` because [this code](https://github.com/openedx/xblock-lti-consumer/blob/f15a78c6054f34411b0a4610083cba91fe5dad7c/lti_consumer/models.py#L567-L584) retrieves the configuration from the external store.

Reply:
What you are suggesting is doing the same has when the lti_config_id parameter is received on the public keyset and access token view but using the external configuration values?. The only issue I think about this is that on the `lti_consumer/api.py` [get_lti_1p3_launch_info](https://github.com/openedx/xblock-lti-consumer/blob/f15a78c6054f34411b0a4610083cba91fe5dad7c/lti_consumer/api.py#L112) function the returned `token_url` and `keyset_url` will be pointing to the current `LtiConfiguration` UUID, in that case a tool deployment for each LTI consumer XBlock will also need to be created, defeating in some way the use of an external LTI configuration, unless `get_lti_1p3_launch_info` returns a `token_url` and `keyset_url` of a common `LtiConfiguration` for all `LtiConfiguration` instances using the same external configuration, I don't think this last approach to be OK for get_lti_1p3_launch_info. Please correct me if I'm wrong on what I understood about your suggestion or the issue I raised is incorrect.